sphinx/ext/autodoc/__init__.py

```diff
@@ -1543,6 +1547,14 @@ def format_signature(self, **kwargs: Any) -> str:
                     documenter.object = func
                     documenter.objpath = [None]
                     sigs.append(documenter.format_signature())
+        if self.analyzer and '.'.join(self.objpath) in self.analyzer.overloads:
+            for overload in self.analyzer.overloads.get('.'.join(self.objpath)):
```

Comment:
No need to do the lookup twice:
```suggestion
        if self.analyzer:
            for overload in self.analyzer.overloads.get('.'.join(self.objpath), []):
```